examples/computer_vision/README.md

```diff
@@ -0,0 +1,160 @@
+# Train and deploy YOLO Object Detection with ZenML, Ultralytics, and FiftyOne
+
+Learn how to build a production-ready computer vision pipeline with YOLOv8, FiftyOne dataset management, and deploy it as a warm HTTP service with an interactive web interface. This example showcases the complete **FiftyOne annotation workflow loop**: export → train → predict → import → analyze → visualize.
+
+## 🎯 What You'll Build
+
+![Interactive Web UI](assets/app.png)
+
+- **Complete FiftyOne Annotation Workflow**: Export COCO data → Train YOLO → Run inference on original FiftyOne dataset → Import predictions back → Analyze performance → Interactive dashboard visualization
+- **Production-Ready Training**: YOLOv8 model training with automatic artifact versioning and performance tracking via ZenML
+- **Real-Time Inference Service**: Deploy as warm HTTP service with sub-second latency using ZenML's deployment system
+- **Interactive Web UI**: Upload images or use URLs for instant object detection testing with visual results
+- **Dataset-Model Lineage**: Full traceability linking FiftyOne datasets to ZenML model artifacts and predictions
+- **Visual Performance Analysis**: Side-by-side comparison of predictions vs ground truth in FiftyOne's interactive dashboard
+
+## 🏃 Quickstart
+
+```bash
+pip install -r requirements.txt
+zenml init
+zenml login
+```
+
+**Train with FiftyOne analysis** ([see code](pipelines/training_pipeline.py)):
+
+```bash
+# Full workflow: training + FiftyOne analysis
+python run.py --train --samples 50 --epochs 3
+
+# Fast training (skip FiftyOne analysis)
+python run.py --train --samples 50 --epochs 3 --disable-fiftyone-analysis
+```
+
+This downloads 50 COCO validation images via FiftyOne, trains a YOLOv8 nano model for 3 epochs, runs inference on the original FiftyOne dataset, and provides interactive analysis capabilities.
+
+**Deploy as a real-time service** ([see code](pipelines/inference_pipeline.py)):
+
+```bash
+zenml pipeline deploy pipelines.inference_pipeline.object_detection_inference_pipeline
+```
+
+Visit `http://localhost:8000` for the interactive UI ([see code](ui/index.html)).
+
+**Test batch inference locally**:
+
+```bash
+python run.py --predict --image https://ultralytics.com/images/bus.jpg
```

Comment:
does this work if yuo have a local image and a pipeline deployed to cloud? I
guess not since it's working with image paths and not raw bytes. I guess this is
why people want the `UploadFile` functionality?